### PR TITLE
Do not append genesis during fork

### PR DESCRIPTION
**Pull request by limebell**
~Instead `IStore.ForkBlockIndexes()` became to iterate from 0 (was 1).~

Instead just append its index only.

### CODE REVIEW

File: Libplanet/Blockchain/BlockChain.cs [hunk 1]
```diff
@@ -174,13 +174,16 @@ IEnumerable<IRenderer<T>> renderers
...
             if (Count == 0)
             {
-                Append(
-                    genesisBlock,
-                    currentTime: genesisBlock.Timestamp,
-                    renderBlocks: !inFork,
-                    renderActions: !inFork,
-                    evaluateActions: !inFork
-                );
+                if (!inFork)
```
Comment (earlbread) [inline Libplanet/Blockchain/BlockChain.cs:177]: How merge if statements?
```suggestion
                if (count == 0 && !inFork)
```

Comment (limebell) [reply Libplanet/Blockchain/BlockChain.cs:177]: It will throw exception in next `elseif` statement 😢

Comment (dahlia) [inline Libplanet/Blockchain/BlockChain.cs:177]: How about merging two conditions into one using an `&&` operator?

**Review by dahlia — APPROVED** [review]

File: Libplanet/Blockchain/BlockChain.cs (hunk 1 above)
Comment (dahlia) [reply Libplanet/Blockchain/BlockChain.cs:177]: Oh, [fair enough](https://github.com/planetarium/libplanet/pull/1063#discussion_r514946332).

Comment (limebell) [reply Libplanet/Blockchain/BlockChain.cs:177]: It will throw exception in next else if statement 😢

**Comment by limebell** [comment]
I made constructor of `BlockChain<T>` instance to not append genesis block every time during fork. Please review again! @libplanet

File: Libplanet/Blockchain/BlockChain.cs [hunk 2]
```diff
@@ -172,7 +172,11 @@ IEnumerable<IRenderer<T>> renderers
                 GetBalance,
                 trieGetter);
 
-            if (Count == 0)
+            if (inFork && !(StateStore is IBlockStatesStore))
```
Comment (longfin) [inline Libplanet/Blockchain/BlockChain.cs:175]: It would be better if we left comments about why we query `StateStore`'s type.

Comment (limebell) [reply Libplanet/Blockchain/BlockChain.cs:175]: 👌

**Review by longfin — APPROVED** [review]

**Review by earlbread — APPROVED** [review]

**Pull request merged**